backend/transcribee_backend/models/task.py

```diff
@@ -83,26 +102,65 @@ class Task(TaskBase, table=True):
     )
 
 
+class TaskAttempt(SQLModel, table=True):
+    id: uuid.UUID = Field(
+        default_factory=uuid.uuid4,
+        primary_key=True,
+        index=True,
+        nullable=False,
+    )
+
+    task_id: uuid.UUID = Field(
+        sa_column=Column(GUID, ForeignKey(Task.id, ondelete="CASCADE"), nullable=False),
+        unique=False,
+    )
+    task: Task = Relationship(
+        back_populates="attempts",
+        sa_relationship_kwargs={
+            "primaryjoin": "TaskAttempt.task_id == Task.id",
+        },
+    )
+
+    assigned_worker_id: Optional[uuid.UUID] = Field(
+        foreign_key="worker.id", default=None
+    )
+    assigned_worker: Optional[Worker] = Relationship()
+
+    started_at: Optional[datetime.datetime] = None
+    last_keepalive: Optional[datetime.datetime] = None
+    ended_at: Optional[datetime.datetime] = None
+
+    progress: Optional[float] = None
+
+    extra_data: Optional[dict] = Field(
+        sa_column=Column(JSON(), nullable=True), default=None
+    )
+
+
+class TaskAttemptResponse(SQLModel):
+    progress: Optional[float]
```

Comment:
that is not a lot of information. Maybe we want to add at least the started time?

Reply:
This is only used for communication with the frontend (this is not a database model). Since we don't currently use this information in the frontend, I'd rather keep it as is and change it once the frontend needs this information